lib/utils.pm

```diff
@@ -273,12 +273,13 @@ sub zypper_call {
     my $ret;
     for (1 .. 3) {
         $ret = script_run("zypper -n $command $printer; ( exit \${PIPESTATUS[0]} )", $timeout);
-        if ($ret == 502) {
-            record_soft_failure "Retrying because of error 502 - bsc#1070851";
-        }
-        else {
-            last;
+        if ($ret == 4) {
+            if (script_run('grep "Error code.*502" /var/log/zypper.log') == 0) {
+                record_soft_failure 'Retrying because of error 502 - bsc#1070851';
+                next;
+            }
         }
+        last;
```

Comment:
According to zypper documentation (http://www.linuxcertif.com/man/8/zypper/#EXIT_CODES_1701h) - 4 stays for "another zypper instance is running" and then you looking for 502 I have a look at the bug but still can't build full picture , can you elaborate ?
Another thing that you basically accepting all other errors which zypper might throw - is this intention ?

Reply:
No, 4 stands for any probllem. And the meego man page is *a bit* outdated.

Reply:
yes it is outdated , but let's take a look to man zypper in my computer ( 1.13.39-14.1  ). All this exit codes can be returned and I think we interested not to skip them  :
 1 - ZYPPER_EXIT_ERR_BUG
  Unexpected situation occurred, probably caused by a bug.
6 - ZYPPER_EXIT_NO_REPOS
  No repositories are defined.
7 - ZYPPER_EXIT_ZYPP_LOCKED
  The ZYPP library is locked, e.g. packagekit is running.
8 - ZYPPER_EXIT_ERR_COMMIT
   An error occurred during installation or removal of packages. You may run zypper verify to repair any dependency problems.
100 - ZYPPER_EXIT_INF_UPDATE_NEEDED
  Returned by the patch-check command if there are patches available for installation.
101 - ZYPPER_EXIT_INF_SEC_UPDATE_NEEDED
    Returned by the patch-check command if there are security patches available for installation.
102 - ZYPPER_EXIT_INF_REBOOT_NEEDED
    Returned after a successful installation of a patch which requires reboot of computer.
104 - ZYPPER_EXIT_INF_CAP_NOT_FOUND
           Returned by the install and the remove command in case any of the arguments does not match any of the available (or installed) package names or other capabilities.
105 - ZYPPER_EXIT_ON_SIGNAL
   Returned upon exiting after receiving a SIGINT or SIGTERM.
107 - ZYPPER_EXIT_INF_RPM_SCRIPT_FAILED
           (Available on SLE-12-SP2/3 and openSUSE-Leap-42.2/3 since 1.13.38 but only if ZYPPER_ON_CODE12_RETURN_107=1 is set in the environment)

Reply:
you right , I haven't seen additional check at the end of function.

Reply:
LGTM